storage_controller/src/reconciler.rs

```diff
@@ -461,6 +487,39 @@ impl Reconciler {
         node: &Node,
         baseline: HashMap<TimelineId, Lsn>,
     ) -> anyhow::Result<()> {
+        // Signal to the pageserver that it should ingest up to the baseline LSNs.
+        loop {
+            match self.wait_lsn(node, tenant_shard_id, baseline.clone()).await {
+                Ok(StatusCode::OK) => {
+                    // Everything is caught up
+                    return Ok(());
+                }
+                Ok(StatusCode::ACCEPTED) => {
+                    // Some timelines are not caught up yet.
+                    // They'll be polled below.
+                    break;
+                }
+                Ok(StatusCode::NOT_FOUND) => {
+                    // None of the timelines are present on the pageserver.
+                    // This is correct if they've all been deleted, but
+                    // let let the polling loop below cross check.
+                    break;
+                }
+                Ok(status_code) => {
+                    tracing::warn!(
+                        "Unexpected status code ({status_code}) returned by wait_lsn endpoint"
+                    );
+                    break;
+                }
+                Err(e) => {
+                    tracing::info!("🕑 Can't trigger LSN wait on {node} yet, waiting ({e})",);
+                    tokio::time::sleep(Duration::from_millis(500)).await;
+                    continue;
+                }
+            }
+        }
+
+        // Poll the LSNs until they catch up
```

Comment:
I guess we still need the polling code below during first upgrade, but can get rid of it later in favor of trusting the pageserver's wait_lsn reply?

Reply:
Polling code should remain. If the LSN does not catch up within the hard-coded 10 seconds, then we fall through from the `StatusCode::ACCEPTED` branch. That's when we'd poll.

The main goal of the new endpoint is to inform the pageserver that _something_ is waiting for it to ingest WAL.
We could set the timeout very low and always poll, but that's wasteful.

Reply:
We could just loop on the new wait lsn endpoint, but each call adds a waiter to the queue on the pageserver side. Perhaps it's not a big deal and we're fine with that.